File: BE/src/main/java/com/example/beprojectweb/entity/Product.java
package com.example.beprojectweb.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long productId;

    @ManyToOne
    @JoinColumn(name = "cate_ID", referencedColumnName = "cate_ID")
    @JsonBackReference
    Category category;
    String productName;
    String description;
    BigDecimal price;
    int stock;
    String img;
<<<<<<< HEAD
=======


>>>>>>> 69320252706cb84eef3f4666440b8312055c7ba3
}

File: FE/src/model/Product.ts
export interface Product {
  productId: number;  // đổi từ product_ID thành productId
  category: Category;  // Tham chiếu đến Category
  productName: string;
  description: string;
  price: number;
  stock: number;
  img: string;
}

export interface ProductRequest {
  productName: string;
  description: string;
  price: number;
  stock: number;
  img: string;
  cate_ID: number;  // giữ nguyên nếu backend yêu cầu
}

export interface ProductResponse {
  productId: number;  // đổi từ product_ID thành productId
  category: Category;
  productName: string;
  description: string;
  price: number;
  stock: number;
  img: string;
}
